client/layout/store-notices/index.js

```diff
@@ -0,0 +1,102 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Component } from '@wordpress/element';
+import { compose } from '@wordpress/compose';
+import PropTypes from 'prop-types';
+import { Notice } from '@wordpress/components';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * WooCommerce dependencies
+ */
+import { OPTIONS_STORE_NAME } from '@woocommerce/data';
+
+/**
+ * Internal dependencies
+ */
+import './style.scss';
+
+class StoreWarnings extends Component {
+	isVisibleMissingPaimentWarning() {
+		const { paymentMethods, queryString } = this.props;
+		const isPaymentPage = queryString.task
+			? queryString.task === 'payments'
+			: false;
+		return ! Object.keys( paymentMethods ).length && isPaymentPage;
+	}
+
+	render() {
+		const showMissingPaimentMethod = this.isVisibleMissingPaimentWarning();
```

Comment:
should be "showMissingPaymentMethod"

Reply:
Nice catch @becdetat, I corrected it in the commit 31d418f

Reply:
@octaedro also `isVisibleMissingPaimentWarning` should be `isVisibleMissingPaymentWarning`